### PR TITLE
Added support for regional calls to GetClusters

### PR DESCRIPTION
**Pull request by cheftako**
**What this PR does / why we need it**:
This adds support for calling GetCluster with Regional clusters.
Also will emit a warning if GetClusters gets a nextPage result. (More than 500 results)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/gce/gce_clusters.go [hunk 1]
```diff
@@ -42,13 +43,19 @@ func (gce *GCECloud) ListClusters(ctx context.Context) ([]string, error) {
...
-		if err != nil {
-			return nil, err
-		}
-		managedClusters = append(managedClusters, clusters...)
+	var location string
+	if len(gce.managedZones) > 1 {
+		// Mutliple managed zones means this is a regional cluster
+		// so use the regional location and not the zone.
+		location = gce.region
+	} else if len(gce.managedZones) == 1 {
```
Comment (lavalamp) [inline pkg/cloudprovider/providers/gce/gce_clusters.go]: return an error if the list is empty?

File: pkg/cloudprovider/providers/gce/gce_clusters.go [hunk 2]
```diff
@@ -42,13 +43,19 @@ func (gce *GCECloud) ListClusters(ctx context.Context) ([]string, error) {
...
+		location = gce.managedZones[0]
 	}
+	clusters, err := gce.getClustersInLocation(location)
+	if err != nil {
+		return nil, err
+	}
+	managedClusters = append(managedClusters, clusters...)
```
Comment (lavalamp) [inline pkg/cloudprovider/providers/gce/gce_clusters.go]: do we need to make managedClusters? Why not just return clusters directly?

File: pkg/cloudprovider/providers/gce/gce_clusters.go [hunk 3]
```diff
@@ -70,13 +77,18 @@ func (gce *GCECloud) listClustersInZone(zone string) ([]string, error) {
 	return result, nil
 }
 
-func (gce *GCECloud) getClustersInZone(zone string) ([]*container.Cluster, error) {
-	mc := newClustersMetricContext("list_zone", zone)
+func (gce *GCECloud) getClustersInLocation(zoneOrRegion string) ([]*container.Cluster, error) {
+	mc := newClustersMetricContext("list_zone", zoneOrRegion)
```
Comment (lavalamp) [inline pkg/cloudprovider/providers/gce/gce_clusters.go]: "list_zone" now seems like an inaccurate name?

File: pkg/cloudprovider/providers/gce/gce_clusters.go (hunk 2 above)
Comment (cheftako) [reply pkg/cloudprovider/providers/gce/gce_clusters.go]: Sorry leftover from the old code. Cleaned up.

File: pkg/cloudprovider/providers/gce/gce_clusters.go (hunk 3 above)
Comment (cheftako) [reply pkg/cloudprovider/providers/gce/gce_clusters.go]: It does. However I am worried about continuity for that metric. Filed https://github.com/kubernetes/kubernetes/issues/68913 to track getting this done as a follow on.

File: pkg/cloudprovider/providers/gce/gce_clusters.go [hunk 4]
```diff
@@ -18,6 +18,9 @@ package gce
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
```
Comment (lavalamp) [inline pkg/cloudprovider/providers/gce/gce_clusters.go]: technically there should be blank lines separating these imports into three sections: std library, internal deps, external deps

File: pkg/cloudprovider/providers/gce/gce_clusters.go [hunk 5]
```diff
@@ -41,24 +44,30 @@ func (gce *GCECloud) ListClusters(ctx context.Context) ([]string, error) {
...
 
-	return managedClusters, nil
+	return clusters, nil
 }
 
 func (gce *GCECloud) Master(ctx context.Context, clusterName string) (string, error) {
 	return "k8s-" + clusterName + "-master.internal", nil
 }
 
 func (gce *GCECloud) listClustersInZone(zone string) ([]string, error) {
```
Comment (lavalamp) [inline pkg/cloudprovider/providers/gce/gce_clusters.go]: This function is not public so hopefully you can fix the name and the parameter name?

Comment (lavalamp) [reply pkg/cloudprovider/providers/gce/gce_clusters.go]: oh wait, I misunderstood. nvm.

File: pkg/cloudprovider/providers/gce/gce_clusters.go (hunk 4 above)
Comment (cheftako) [reply pkg/cloudprovider/providers/gce/gce_clusters.go]: Fixed.

**Pull request merged**